src/iter.rs

```diff
@@ -1,7 +1,8 @@
 use std::io;
+use std::iter::Peekable;
 
 pub struct LineColIterator<Iter: Iterator<Item=io::Result<u8>>> {
-    iter: Iter,
+    iter: Peekable<Iter>,
```

Comment:
Rather than making this always Peekable, could you add an impl for it to be peekable if the underlying iterator is peekable?